src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -249,10 +259,14 @@ private function getCartProductData(Cart $cart, Product $product, ?Combination $
         });
         $productItem['cart_quantity'] = $quantity;
 
+        $productItem['price_with_reduction'] = (float) (string) $command->getProductPriceTaxIncluded();
+        $productItem['price_without_reduction'] = (float) (string) $command->getProductPriceTaxIncluded();
+        $productItem['price_wt'] = (float) (string) $command->getProductPriceTaxIncluded();
+
         switch (Configuration::get('PS_ROUND_TYPE')) {
             case Order::ROUND_TOTAL:
                 $productItem['total'] = $productItem['price_with_reduction_without_tax'] * $quantity;
-                $productItem['total_wt'] = $productItem['price_with_reduction'] * $quantity;
+                $productItem['total_wt'] = (float) (string) $command->getProductPriceTaxIncluded() * $quantity;
```

Comment:
You should be able to use `$productItem['price_without_reduction']` instead since you already replaced it correctly, this would we more consistent because you already use `$productItem['price_with_reduction_without_tax']`

And it avoids performing the type casting each time